datafusion/core/src/physical_plan/planner.rs

```diff
@@ -2518,7 +2518,8 @@ Internal error: Optimizer rule 'type_coercion' failed due to unexpected error: E
             match ctx.read_csv(path, options).await?.into_optimized_plan()? {
                 LogicalPlan::TableScan(ref scan) => {
                     let mut scan = scan.clone();
-                    scan.table_name = name.to_string();
+                    let table_reference: TableReference<'_> = name.into();
+                    scan.table_name = table_reference.to_owned_reference();
```

Comment:
```suggestion
                    let table_reference = TableReference::from(name).to_owned_reference();
                    scan.table_name = table_reference;
```

maybe this possibly?

Reply:
That looks much better -- thank you 👍